src/index.ts

```diff
@@ -296,10 +324,21 @@ export class Upload extends Pumpify {
       configPath,
     });
 
+    const autoRetry = cfg?.retryOptions?.autoRetry || AUTO_RETRY_VALUE;
     this.uriProvidedManually = !!cfg.uri;
     this.uri = cfg.uri || this.get('uri');
     this.numBytesWritten = 0;
     this.numRetries = 0;
+    this.retryLimit = autoRetry
+      ? cfg?.retryOptions?.maxRetries || RETRY_LIMIT
+      : 0;
+    this.maxRetryDelay = cfg?.retryOptions?.maxRetryDelay || MAX_RETRY_DELAY;
+    this.retryDelayMultiplier =
+      cfg?.retryOptions?.retryDelayMultiplier || RETRY_DELAY_MULTIPLIER;
+    this.maxRetryTotalTimeout =
+      cfg?.retryOptions?.totalTimeout || MAX_TOTAL_RETRY_TIMEOUT;
```

Comment:
I believe this needs to confirm a number wasn't provided, in case a user provides a `0`. e.g. `0 || MAX_RETRY_DELAY` would resolve to `MAX_RETRY_DELAY` and ignore the user's input.